daemon/create.go

```diff
@@ -47,7 +48,7 @@ func (daemon *Daemon) ContainerCreate(job *engine.Job) engine.Status {
 
 	container, buildWarnings, err := daemon.Create(config, hostConfig, name)
 	if err != nil {
-		if daemon.Graph().IsNotExist(err) {
+		if daemon.Graph().IsNotExist(err) && strings.Contains(err.Error(), config.Image) {
```

Comment:
I think just as you do. Fix `IsNotExist` is the first way I thought of,  `IsNotExist` is weak, but I worry about fix `IsNotExist` will cause a lot of modification since it is a utility, I thought it was a not suggested way.  
But since it was a pain, we have to fix it, I'll try to fix `IsNotExist`  :)